### PR TITLE
[BUG] Batch selection disappears before user takes action

### PR DESCRIPTION
**Pull request by nambatee**
https://phabricator.wikimedia.org/T208370

### CODE REVIEW

File: WMF Framework/CollectionViewEditController.swift [hunk 1]
```diff
@@ -613,9 +613,11 @@ public class CollectionViewEditController: NSObject, UIGestureRecognizerDelegate
     }
     
     @objc public func didPerformBatchEditToolbarAction(with sender: UIBarButtonItem) {
-        let didPerformAction = delegate?.didPerformBatchEditToolbarAction?(batchEditToolbarActions[sender.tag]) ?? false
-        if didPerformAction {
-            editingState = .closed
+        let action = batchEditToolbarActions[sender.tag]
+        delegate?.didPerformBatchEditToolbarAction?(action) { finished in
+            if finished {
+                self.editingState = .closed
```
[the remark this answers is not in the document]
Comment (joewalsh) [reply WMF Framework/CollectionViewEditController.swift]: Also, should the batch selection stay active if the user taps "Cancel" on the alert? Currently it looks like it deselects and closes batch selection

[the remark this answers is not in the document]
Comment (nambatee) [reply WMF Framework/CollectionViewEditController.swift]: sure, good idea 👍

[the remark this answers is not in the document]
Comment (nambatee) [reply WMF Framework/CollectionViewEditController.swift]: good question 🤔 confirming with Carolyn re the expected behavior

[the remark this answers is not in the document]
Comment (nambatee) [reply WMF Framework/CollectionViewEditController.swift]: confirmed with Carolyn and updated to keep it active when user taps "Cancel" 👍

[the remark this answers is not in the document]
Comment (nambatee) [reply WMF Framework/CollectionViewEditController.swift]: also confirmed with Carolyn we want to keep the action pane open when user taps "Cancel", updated

**Pull request merged**